src/main/kotlin/org/pkl/intellij/packages/PklPackageService.kt

```diff
@@ -132,6 +132,19 @@ class PklPackageService(val project: Project) : Disposable, UserDataHolderBase()
     return projectPackages + declaredPackages
   }
 
+  fun downloadPackage(packages: List<PackageUri>): CompletableFuture<Unit> {
+    return CompletableFuture<Unit>().apply {
+      runBackgroundableTask("Download packages") {
```

Comment:
Isn't it a Future already a background task? Does it play well with IntelliJ's api?

Reply:
No; it's an abstraction that allows you to represent async values, but itself is not a task.

The JDK provides `FutureTask` that can be paired with an `Executor` to execute background tasks.
But, there's no need to use them because we already have `runBackgroundableTask` from the IJ SDK.
For cases like this, `CompletableFuture` exists to allow us to turn some async callback-based API into a Future value.